### ISSUE TITLE
Add an include_details query param to the API

### ISSUE THREAD

**Issue by andreidan** [elastic/elasticsearch#86215]
As the API is designed for user consumption (but can also be polled by various monitoring tools)
we have to balance the amount of work the API does in terms of RCA vs returning as much information 
as possible (ie. the details)

This proposes the introduction of an `include_details` query parameter that defaults to `true`.

This will be changed to `false` when the API is polled, but the users can simply call `_internal/_health` to
retrieve all the details.

On drill down (when the details are included by virtue of drilling down ie. `_internal/_health/data`) if
`include_details` contradicts the default drill down behaviour ( ie. `_internal/_health/data?include_details=false`) the 
details will be ommited (ie. `include_details` overrides the drill down behaviour)

**Comment by masseyke** [comment]
> This proposes the introduction of an `include_details` query parameter that defaults to `true`.

You're proposing changing even the default behavior of  `_internal/_health` to include details by default, right? (I think that's what you're saying, just wanted to confirm).

**Comment by andreidan** [comment]
@masseyke Yes, that's the desire - to prioritise user consumption as opposed to "machine"/polling consumption
